### ISSUE TITLE
Trying to make a multiligual site

### ISSUE THREAD

**Issue by CptCompiler** [wpengine/faustjs#343]
Hey,
first of all thank you for this super cool framework.

I've installed wp polylang and the wp polylang graphql plugins.
I configured next to use two languages english and german.

next.js says `For pages that are automatically statically optimized, a version of the page will be generated for each locale.`

Everything works okay. The statically generated html files look correct.
But the JSON files containing the page props are missing the content from the api but only for one of the two languages.

I don't really know what's going wrong here. 
I just noticed that calling the `prepareReactRender` (in your  in your getProps function) twice before returning seems to work 🤷‍♂️

do you have any clue how to solve this issue?

**Comment by blakewilson** [comment]
Hey @CptCompiler, thanks for the nice words!

Are you using the `wpengine/headless-next` package or `faustjs/headless-next`?

Additionally, could you paste in the code for the  [Next.js page](https://nextjs.org/docs/basic-features/pages) you see this occurring on?

**Comment by CptCompiler** [comment]
Hi @blakewilson,

thank's for answering.

I am using faustjs. I have forked the repository and changed the `next/getting-started` project.
[here](https://github.com/CptCompiler/headless-framework/commit/0b51e2740db11d7885247047f9409dce1eef535b)

- I have added a language switch to navigate to the other language on the client side.
- added two languages via next.config.js
- added the locale to the `getNextStaticProps`
```
export async function getStaticProps(context: GetStaticPropsContext) {
  return getNextStaticProps(context, {
    Page,
    client,
    props: {
      locale: context.locale
    }
  });
}
```

- in the post index page ich changed the where statement for fetching posts, so that if the language is "de" first 2 posts should be fetched and when the language is "en" the last 2 posts should be fetched

```const lang = props.locale || router.locale
  let where = {}
  if (lang === "de") {
    where = {
      first: 2
    }
  }
  else {
    where = {
      last: 2
    }
  }
  const posts = usePosts(where);
```

server side rendering is working good. If I serve the `/posts` and `/en/posts` urls directly, both pages show the correct two posts.
In my case the generated JSON file for `en` is empty. So if I `npm run build` and `npm run start` and go directly to `/posts` and then navigate via the language switch to `/en/posts` the rendered post teasers do not change. (If there are no posts in the en JSON file I would have guessed that no posts would show up. But thats not the case, the posts from the de language are still visible)

After some navigating and page reloading everything works (JSON files are regenerated correctly). I think thats because of the `Incremental Static Regeneration` ???

**Comment by wjohnsto** [comment]
First of all, I do not believe you are doing anything wrong :). Second, I am working through an issue that I think is similar to this where I have two pages that are SSG (or ISR I guess), and both use a variation of the same GQL query. One gets the first 3 posts and the other gets a list of posts. What I experience is eventual consistency, meaning that after building the site the JSON file is empty, but after visiting the page on the frontend and reloading the page they eventually become correct. It seems to be some issue with how gqless query caching/`prepareReactRender` works with Next's SSG. I'm going to try to get a simple project setup in an attempt to isolate the issue. Stay tuned.

**Comment by wjohnsto** [comment]
@CptCompiler We just released v0.7.2 of all the packages, update to that and see if it fixes your issue.

**Comment by wjohnsto** [comment]
@CptCompiler I believe I have identified the issue in `gqless` and offered a solution here: https://github.com/gqless/gqless/issues/237
